Sming/Wiring/WVector.h

```diff
@@ -119,14 +119,14 @@ Vector<Element>::Vector(unsigned int initialCapacity, unsigned int capacityIncre
   _capacity = initialCapacity;
   _data = new Element*[ _capacity ];
   _increment = capacityIncrement;
-  if (_data == NULL) _capacity = _increment = 0;
-};
+  if (_data == nullptr) _capacity = _increment = 0;
+}
```

Comment:
Can you leave the closing semicolon? This way the files in Wiring will be consistent (see also WHashMap for example).  Or was that change due to a warning reported from the compiler?

Reply:
It's only for method implementations, not the class definition - WHashMap does everything inline so unless I'm missing something it's entirely consistent. I removed them because autoformat makes it look very strange (although we're not doing that, at least for now), no compiler warnings.

Reply:
> It's only for method implementations, not the class definition

You are absolutely right.  Now comes the question how to solve the quite-so-often error of creating a (template) class definition without the trailing semicolon :). There are two major ways to solve that
- we remove all semicolons from template method definitions (which is correct C++). And spend time educating every newcomer that made the mistake. 
- we put semicolons everywhere on both template classes and template methods. This way, hopefully, there will be less explaining to do.

I would prefer the first solution. But that should be a part of a separate PR where also the other template definitions are revised and consistent. If you agree - please put back the semicons here and create a new PR where they are removed from the template method definitions. 

What do you say?

Reply:
No probs, I'll put the semicolons back!

Reply:
It's pretty simple rule: no semicolons after function (or method) definition :-)

Reply:
> We love those semicolons!

But not for long. The new PR will fix this, right :)

Reply:
Natürlich. I've had a quick look through and as far as I can tell WVector.h is the only offender. Do you know of any others?

Reply:
> Do you know of any others?

I ran quick reg-ex search over the code and only WVector.h showed up. Which is good.